src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,26 +105,48 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
+  }, [numberOfImgs, randomPhotos]);
+
+  type unsplashPhotoFortmat = {
+    id: string;
+    url: string;
+    alt: string;
+    urls: { regular: string };
+    alt_description: string;
+  };
+
+  /*----------------------FETCH RANDOM IMAGES ---------------------- */
 
-  // to avoid bug undefined url when user change number of images
   useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
+    async function getData() {
```

Comment:
Generally, I'd prefer to see the API requests separated from Components (we want to keep the project organized from the very beginning:)). One good pattern might be using hooks.
Here's a small and simple example from a test project I did a while time ago:
https://github.com/sashabaranova/easybanking/tree/main/src/hooks